Objects/dictobject.c

```diff
@@ -2358,17 +2359,31 @@ _PyDict_LoadGlobal(PyDictObject *globals, PyDictObject *builtins, PyObject *key)
             return NULL;
     }
 
+#ifdef Py_GIL_DISABLED
     /* namespace 1: globals */
-    ix = _Py_dict_lookup(globals, key, hash, &value);
+    ix = _Py_dict_lookup_threadsafe(globals, key, hash, &value);
```

Comment:
I think this pattern might be simpler if we define `_Py_dict_lookup_threadsafe` in the default build as `_Py_dict_lookup_threadsafe` + `Py_XNewRef()`.

We could use it in `dict_subscript` and `dict_get_impl` as well.